models/Asset/Document/ImageThumbnail.php

```diff
@@ -95,12 +95,12 @@ public function generate($deferredAllowed = true)
 
                     $lock = \Pimcore::getContainer()->get(LockFactory::class)->createLock('document-thumbnail-' . $this->asset->getId() . '-' . $this->page);
 
-                    if (!is_file($path) && !$lock->isAcquired()) {
-                        $lock->acquire();
+                    $lockAcquired = $lock->acquire();
+                    if (!is_file($path) && $lockAcquired) {
                         $converter->saveImage($path, $this->page);
                         $generated = true;
                         $lock->release();
-                    } elseif ($lock->isAcquired()) {
+                    } elseif (!$lockAcquired) {
```

Comment:
I would here use `$lock->isAcquired()` and let the `$lock->acquire()` behind `!is_file($path)`

Should not be locked if the path exists. Only if the file is missing

Reply:
@blankse But `isAcquired()` only works for the same process, we actually want to process only one document at the time here, so we have to use `acquire()`

Reply:
Is the same process in the if above

Reply:
Sorry, I don't get your point😐
It's an `if` /  `elseif` so if we got the lock in this process ( by `$lockAcquired = $lock->acquire();`) we never reach the `elseif` anyway, so this is just for the case that we're in another process where we didn't got the lock because the conversion is already in progress by another process.

Reply:
Okay, I get your point. The lock was also selected before if the file exists. I think this is a unnecessary SQL. Why not make 2 ifs?
```
if (!is_file($path)) {
    if ($lock->acquire()) {
        ...
    } else {
       ...
    }
}
```

Reply:
Or can exist the file but is not ready... Then we should leave it as it is.

Reply:
Yep - done - thanks! 😊